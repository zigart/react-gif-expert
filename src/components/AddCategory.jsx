import { useState } from "react";
import Proptypes from "prop-types";
export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");
  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();

    const newInputValue = inputValue.trim();
    if (newInputValue.length <= 1) return;

    onNewCategory(newInputValue);
    setInputValue("");
  };
  return (
    <form aria-label="form" onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
      ></input>
    </form>
  );
};

AddCategory.prototype = {
  onNewCategory: Proptypes.func.isRequired,
};
