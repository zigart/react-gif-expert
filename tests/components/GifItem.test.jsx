import { AddCategory, GifItem } from "../../src/components";
import { render, screen } from "@testing-library/react";

describe("Pruebas del componente <GifItem />", () => {
  const title = "Dragon Ball";
  const url = "https://dragonball.gif/";
  test("El render coincide con el snapshot ", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
  test("Debe mostrar la imagen con la URL y el ALT indicado", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    const { src, alt } = screen.getByRole("img");

    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("Debe mostrar el titulo en el componente", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
