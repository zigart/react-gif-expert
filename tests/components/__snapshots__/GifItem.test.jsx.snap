// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifItem /> El render coincide con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball"
      src="https://dragonball.gif/"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
